receiver/mongodbatlasreceiver/alerts.go

```diff
@@ -204,6 +343,97 @@ func (a alertsReceiver) Shutdown(ctx context.Context) error {
 	return nil
 }
 
+func (a alertsReceiver) shutdownRetriever(ctx context.Context) error {
+	a.logger.Debug("Shutting down client")
+	a.doneChan <- true
+	a.checkpointDoneChan <- true
```

Comment:
I think the "idomatic" way to do this is to just close these channels
```suggestion
	close(a.doneChan)
	close(a.checkpointDoneChan)
```